crates/ruff_python_ast/ast.toml

```
@@ -80,39 +80,367 @@ add_suffix_to_is_methods = true
+name = "elts"
+type = "Expr"
+seq = true
+
+[Expr.nodes.ExprListComp]
+rustdoc = "/// See also [ListComp](https://docs.python.org/3/library/ast.html#ast.ListComp)"
+
+[[Expr.nodes.ExprListComp.fields]]
+name = "elt"
+type = "Expr"
+
+[[Expr.nodes.ExprListComp.fields]]
+name = "generators"
+type = "Comprehension"
+seq = true
+
+[Expr.nodes.ExprSetComp]
+rustdoc = "/// See also [SetComp](https://docs.python.org/3/library/ast.html#ast.SetComp)"
+
+[[Expr.nodes.ExprSetComp.fields]]
+name = "elt"
+type = "Expr"
+
+[[Expr.nodes.ExprSetComp.fields]]
+name = "generators"
+type = "Comprehension"
+seq = true
+
+[Expr.nodes.ExprDictComp]
+rustdoc = "/// See also [DictComp](https://docs.python.org/3/library/ast.html#ast.DictComp)"
+
+[[Expr.nodes.ExprDictComp.fields]]
+name = "key"
+type = "Expr"
+
+[[Expr.nodes.ExprDictComp.fields]]
+name = "value"
+type = "Expr"
+
+[[Expr.nodes.ExprDictComp.fields]]
+name = "generators"
+type = "Comprehension"
+seq = true
+
+[Expr.nodes.ExprGenerator]
+rustdoc = "/// See also [GeneratorExp](https://docs.python.org/3/library/ast.html#ast.GeneratorExp)"
+
+[[Expr.nodes.ExprGenerator.fields]]
+name = "elt"
+type = "Expr"
+
+[[Expr.nodes.ExprGenerator.fields]]
+name = "generators"
+type = "Comprehension"
+seq = true
+
+[[Expr.nodes.ExprGenerator.fields]]
+name = "parenthesized"
+type = "bool"
+
+[Expr.nodes.ExprAwait]
+rustdoc = "/// See also [Await](https://docs.python.org/3/library/ast.html#ast.Await)"
+
+[[Expr.nodes.ExprAwait.fields]]
+name = "value"
+type = "Expr"
+
+[Expr.nodes.ExprYield]
+rustdoc = "/// See also [Yield](https://docs.python.org/3/library/ast.html#ast.Yield)"
+
+[[Expr.nodes.ExprYield.fields]]
+name = "value"
+type = "Expr"
+optional = true
+
+[Expr.nodes.ExprYieldFrom]
+rustdoc = "/// See also [YieldFrom](https://docs.python.org/3/library/ast.html#ast.YieldFrom)"
+
+[[Expr.nodes.ExprYieldFrom.fields]]
+name = "value"
+type = "Expr"
+
+[Expr.nodes.ExprCompare]
+rustdoc = "/// See also [Compare](https://docs.python.org/3/library/ast.html#ast.Compare)"
+
+[[Expr.nodes.ExprCompare.fields]]
+name = "left"
+type = "Expr"
+
+[[Expr.nodes.ExprCompare.fields]]
+name = "ops"
+type = "Box<[crate::CmpOp]>"
+
+[[Expr.nodes.ExprCompare.fields]]
+name = "comparators"
+type = "Box<[Expr]>"
+
+[Expr.nodes.ExprCall]
+rustdoc = "/// See also [Call](https://docs.python.org/3/library/ast.html#ast.Call)"
+
+[[Expr.nodes.ExprCall.fields]]
+name = "func"
+type = "Expr"
+
+[[Expr.nodes.ExprCall.fields]]
+name = "arguments"
+type = "Arguments"
+
+[Expr.nodes.ExprFString]
+rustdoc = """/// An AST node that represents either a single-part f-string literal
+/// or an implicitly concatenated f-string literal.
+///
+/// This type differs from the original Python AST ([JoinedStr]) in that it
+/// doesn't join the implicitly concatenated parts into a single string. Instead,
+/// it keeps them separate and provide various methods to access the parts.
+///
+/// [JoinedStr]: https://docs.python.org/3/library/ast.html#ast.JoinedStr"""
```

Nit: We could also use a list here and join the elements?
```suggestion
rustdoc = [
	"An AST node that represents either a single-part f-string literal",
	"or an implicitly concatenated f-string literal.",
	"",
	"This type differs from the original Python AST ([JoinedStr]) in that it",
	"doesn't join the implicitly concatenated parts into a single string. Instead,",
	"it keeps them separate and provide various methods to access the parts.",
	"",
	"[JoinedStr]: https://docs.python.org/3/library/ast.html#ast.JoinedStr"
]
```

But I must admit it looks worse with the empty lines... hmm

I would definetely remove the need for `///` They're unnecessary noise when authoring the TOML file and they can easily be inserted by the script.

---

That's a good point that the script can insert the `///`s.  I think removing those, but keeping the multi-line string, is likely to be a good compromise:

```suggestion
rustdoc = """
An AST node that represents either a single-part f-string literal
or an implicitly concatenated f-string literal.

This type differs from the original Python AST ([JoinedStr]) in that it
doesn't join the implicitly concatenated parts into a single string. Instead,
it keeps them separate and provide various methods to access the parts.

[JoinedStr]: https://docs.python.org/3/library/ast.html#ast.JoinedStr"""
```

---

Good point I'm going with the second version with multi line strings.